jenkins/e2e-image/e2e-runner.sh

```diff
@@ -289,28 +290,35 @@ EXIT_CODE=0
 
 # We get the Kubernetes tarballs unless we are going to use old ones
 if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
-    echo "Using existing binaries; not cleaning, fetching, or unpacking new ones."
+  echo "Using existing binaries; not cleaning, fetching, or unpacking new ones."
 else
-    clean_binaries
-
-    if [[ "${JENKINS_USE_LOCAL_BINARIES:-}" =~ ^[yY]$ ]]; then
-        set_release_vars_from_local_gcs_stage
-    elif [[ "${JENKINS_USE_SERVER_VERSION:-}" =~ ^[yY]$ ]]; then
-        # This is for test, staging, and prod jobs on GKE, where we want to
-        # test what's running in GKE by default rather than some CI build.
-        set_release_vars_from_gke_cluster_version
-    elif [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
-        # Use GCI image builtin version. Needed for GCI release qual tests.
-        set_release_vars_from_gci_builtin_version
-    else
-        # use JENKINS_PUBLISHED_VERSION, default to 'ci/latest', since that's
-        # usually what we're testing.
-        set_release_vars_from_gcs "${JENKINS_PUBLISHED_VERSION:-ci/latest}"
-        # Needed for GKE CI.
-        export CLUSTER_API_VERSION=$(echo "${KUBERNETES_RELEASE}" | cut -c 2-)
-    fi
+  clean_binaries
+
+  if [[ "${JENKINS_SOAK_MODE:-}" == "y" && ${E2E_UP:-} != "true" ]]; then
+    # We are restoring the cluster, copy state from gcs
+    # TODO(fejta): auto-detect and recover from deployment failures
+    gsutil cp "${HOME}/.kube/config" "${JENKINS_SOAK_BUCKET}/${JOB_NAME}/kube-config"
```

Comment:
might also need to `mkdir -p` the `.kube` dir, since I don't know if gsutil will do that for you.